### DISCUSSION TITLE
Bulk Insert on Initial Sync

### DISCUSSION THREAD

**Discussion by Arslan007** [Azure/azure-mobile-apps#699] [Q&A]
We are using this wonderful library in our project and we are loving it for every bit of it.
Only question we have is in our case when we do initial sync then one table have lets say 10K records and each records is saved in sqlite one by one and it take so much time for initial load. Is there anyway we can do bulk insert on initial sync as that time we will have no conflicts and no extra checking will be required that time. 
Any suggestions in this regard will be appreciated. 

Thanks!

**Comment by adrianhall** [comment]
With the 5.1.6 release, the records are inserted in bulk according to the "WriteDeltaTokenInterval".  You want to go through the pull so that the delta token is updated.  However, you can set the delta-token interval to a relatively high number (e.g. 100) so that they are written in bulk.

Something like:

```csharp
await table.PullItemsAsync(new PullOptions { WriteDeltaTokenInterval = 100 });
```

This is a trade-off between database consistency and speed.  You get to choose which you prioritize.
